pkg/koordlet/metriccache/tsdb_storage.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metriccache
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/prometheus/model/labels"
+	promstorage "github.com/prometheus/prometheus/storage"
+	"github.com/prometheus/prometheus/tsdb"
+	"k8s.io/klog/v2"
+)
+
+// TSDBStorage defines time-series type DB, providing insert and query interface
+type TSDBStorage interface {
+	Appendable
+	Queryable
+
+	// Close closes the storage and all its underlying resources.
+	Close() error
+}
+
```

Comment:
how to truncate the old data intsdb storage?

Reply:
https://github.com/koordinator-sh/koordinator/pull/1228/files#diff-45a081cdd0ea16350b2cdeb04da7a47f747a809739ef2c1ece292b4e1ec9a89dL26-L31

RetentionDuration and MaxBytes options are supported by prometheus.tsdb.